### DISCUSSION TITLE
The "Resize" operation is dangerous

### DISCUSSION THREAD

**Discussion by DCNick3** [WerWolv/ImHex#1463] [Feedback]
The "Resize" operation operates directly on the underlying file (for efficiency reasons, I presume), which breaks my expectations of "nothing is changed until I press save".

I actually used it by accident from the Right-click menu thinking that it was for changing width and instantly garbled up the file I was looking at. (I later found the unmarked slider saying "16" in the bottom right that actually does this).

What also makes the menu not nice (and contributed to my misunderstanding of it as a width-changing command) is the fact that it doesn't fill in the original size of the file, instead just being `0x0` initially.

I guess what could have helped is an additional warning before running the resize? Maybe also rename it to "Resize file" for less confusion? (less confident about that)

**Comment by paxcut** [comment]
thats odd. i tried to make the size of the rom file you posted in the other issue smaller and it got smaller, but not to the size i specified. I only use resize when i create new files which are always sized zero.  While I agree it is potentially dangerous and a warning about possible data destruction should be popping out  there is an undo menu item that will recover the file as it was originally. Maybe the warning can also state that as a reminder. This is not something I can fix though, just sharing my opinion.

**Reply by DCNick3** [reply]
Undo doesn't seem to be able to cancel that

(maybe because my file is 7 GiB :sweat_smile:)

**Reply by paxcut** [reply]
Hold on, was the file changed in the hard drive without you saving it from the file menu?

**Reply by DCNick3** [reply]
Yes, this is what happens when I resize the file: it is modified directly

**Reply by paxcut** [reply]
no. then that is beyond dangerous. if it erases data from the disk and it is unable to undo the change then it should not be available for file providers and used only on memory providers

**Reply by Nemoumbra** [reply]
Y'all are talking about the latest release, no doubt. The nightly's got a better undo/redo stack that supports undoing the resize action.

**Reply by paxcut** [reply]
no, not me, my undo works fine after resizing. but my file is not 7 gbs. we thought maybe that was the problem but if Nick is using release then that would explain it.

**Reply by DCNick3** [reply]
That is great to hear! I'll try out the newer version then

**Reply by Nemoumbra** [reply]
Warning, if you resize your 7 Gb file to just a few bytes in Nightly, it'll save the deleted part to RAM. It's still a dangerous operation, but in a different way now.

@WerWolv maybe we should show a warning if the deleted size is too big? 0.5 Gb should be enough to defend people against such feature misuses.

**Reply by DCNick3** [reply]
I do notice such a behavior (it does seem to add 7 GiB to the memory consumption each time I resize), but, confusingly, the bytes I see in the hex editor don't change the size? The saving also doesn't seem to be causing the file size to change.. Perhaps I should open this as an issue then? :sweat_smile: 

(Not that I particularly care about resizes, but it seems to just have stopped doing what it should altogether)

**Reply by paxcut** [reply]
in my case i had a file that was 63 ks. if i resized and asked for 16 bytes the file would be truncated to about half its size. undo would recover the lost half.

**Reply by WerWolv** [reply]
Fixed the issue with the resize operation not working correctly
https://github.com/WerWolv/ImHex/commit/b1aa4fd3f87e7f9088692f75680a5422c0e27105

About the warning, I'll think about it. We need a better system to issue these warnings first
